angular/projects/admin-nrpti/src/app/sidebar/sidebar.component.ts

```diff
@@ -37,15 +39,51 @@ export class SidebarComponent implements OnDestroy {
       });
   }
 
+  // SetActiveSidebarItem() {
+  //   const urlArray = this.routerSnapshot.url.split('/');
+  //   // urlArray[0] will be empty so we use shift to get rid of it.
+  //   urlArray.shift();
+  //   this.currentMenu = urlArray[0];
+  // }
+
   SetActiveSidebarItem() {
     const urlArray = this.routerSnapshot.url.split('/');
-    // urlArray[0] will be empty so we use shift to get rid of it.
+
+    // The first element will be empty, so shift in order to remove it.
     urlArray.shift();
-    this.currentMenu = urlArray[0];
-  }
+    const [mainRoute, mainRouteId, currentMenu] = urlArray;
+
+    this.mainRoute = mainRoute;
+    this.mainRouteId = mainRouteId;
+    this.currentMenu = currentMenu && currentMenu.split(';')[0];
+
+    switch (mainRoute) {
```

Comment:
Nit: Could probably just be an `if`, rather than a `switch` with only one `case`?

edit: Looks like you could probably merge it with the `if (mainRoute === 'mines') {` on line 72?